ignite/handlers/lr_finder.py

```diff
@@ -103,12 +103,10 @@ def _run(
             num_iter = trainer.state.epoch_length * trainer.state.max_epochs
         else:
             max_iter = trainer.state.epoch_length * trainer.state.max_epochs  # type: ignore[operator]
-            if num_iter > max_iter:
-                warnings.warn(
-                    f"Desired num_iter {num_iter} is unreachable with the current run setup of {max_iter} iteration "
-                    f"({trainer.state.max_epochs} epochs)",
-                    UserWarning,
-                )
+            if max_iter < num_iter:
+                max_iter = num_iter
+                trainer.state.max_iters = num_iter
+                trainer.state.max_epochs = num_iter // trainer.state.epoch_length  # type: ignore[operator]
```

Comment:
I wonder if this should not be ceil instead of floor...

Reply:
I think it should stay floor, for example 150 num_iters should be 1 epoch I think, and have a look here here for example https://github.com/pytorch/ignite/blob/master/tests/ignite/handlers/test_lr_finder.py#L306

Reply:
Thanks for the link. I think this test shows that the API is a bit weird:
- dataloader size is 100
- user requires 150 iters to do
```python
    with lr_finder.attach(dummy_engine, to_save, num_iter=150, diverge_th=float("inf")) as trainer_with_finder:
        with pytest.warns(UserWarning):
            trainer_with_finder.run(dataloader)
        # We should do 150 iters and not the size of the dataloader:
        assert dummy_engine.state.iteration != len(dataloader)
        assert dummy_engine.state.iteration == 150
```

Reply:
Thanks for the clarification, you're right, this test should be edited according to the updates, as the lrfinder now is able to run more than one epoch, will edit it